hack/observability/kube-state-metrics/crd-config.yaml

```diff
@@ -753,3 +811,176 @@ spec:
+      cluster_name:
+      - spec
+      - clusterName
+      name:
+      - metadata
+      - name
+      namespace:
+      - metadata
+      - namespace
+      uid:
+      - metadata
+      - uid
+    metricNamePrefix: capi_machinepool
+    metrics:
+      - name: spec_replicas
+        help: The number of desired machines for a machinepool.
+        each:
+          gauge:
+            path:
+            - spec
+            - replicas
+          type: Gauge
+      - name: status_replicas
+        help: The number of replicas per machinepool.
+        each:
+          gauge:
+            path:
+            - status
+            - replicas
+            nilIsZero: true
+          type: Gauge
+      - name: status_replicas_ready
+        help: The number of ready replicas per machinepool.
+        each:
+          gauge:
+            path:
+            - status
+            - readyReplicas
+            nilIsZero: true
+          type: Gauge
+      - name: status_replicas_available
+        help: The number of available replicas per machinepool.
+        each:
+          gauge:
+            path:
+            - status
+            - availableReplicas
+            nilIsZero: true
+          type: Gauge
+      - name: status_replicas_unavailable
+        help: The number of unavailable replicas per machinepool.
+        each:
+          gauge:
+            path:
+            - status
+            - unavailableReplicas
+            nilIsZero: true
+          type: Gauge
+      - name: info
+        each:
+          type: Info
+          info:
+            labelsFromPath:
+              infrastructure_reference_name:
+              - spec
+              - template
+              - spec
+              - infrastructureRef
+              - name
+              infrastructure_reference_kind:
+              - spec
+              - template
+              - spec
+              - infrastructureRef
+              - kind
+              bootstrap_configuration_reference_name:
+              - spec
+              - template
+              - spec
+              - bootstrap
+              - configRef
+              - name
+              bootstrap_configuration_reference_kind:
+              - spec
+              - template
+              - spec
+              - bootstrap
+              - configRef
+              - kind
+              failure_domain:
+              - spec
+              - template
+              - spec
+              - failureDomain
+              version:
+              - spec
+              - template
+              - spec
+              - version
+      - name: status_phase
+        help: The machinepools current phase.
+        each:
+          stateSet:
+            labelName: phase
+            list:
+            - ScalingUp
+            - ScalingDown
+            - Running
+            - Failed
+            - Unknown
+            path:
+            - status
+            - phase
+          type: StateSet
+    - name: created
```

Comment:
wrong indent beginning here?! or tabs vs spaces?

Reply:
"wrong" is relative ;-)

i use [the `YAML` plugin in vscode](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml) which auto-format the yaml file. but anyhow - i fixed the format to allign with the existing generated yamls.

Reply:
uh, then the others may be wrong :-) happy to get it auto-formatted but also okay to keep as is :-)

Sad that there is not the only formatter for yaml, like there is gofmt for go 😂